src/Compilers/CSharp/Test/Semantic/Semantics/StaticNullChecking_Members.cs

```diff
@@ -0,0 +1,437 @@
+class B
+{
+    internal A? A;
+}
+class C
+{
+    internal B? B;
+}
+class Program
+{
+    static void F()
+    {
+        object o;
+        C? c = new C();
+        c.B = new B();
+        c.B.A = new A();
+        o = c.B.A; // 1
+        c.B.A = null;
+        o = c.B.A; // 2
+        c.B = new B();
+        o = c.B.A; // 3
+        c.B = null;
+        o = c.B.A; // 4
+        c = new C();
+        o = c.B.A; // 5
+        c = null;
+        o = c.B.A; // 6
+    }
+}";
+            var comp = CreateStandardCompilation(source, parseOptions: TestOptions.Regular8);
+            comp.VerifyDiagnostics(
+                // (24,13): warning CS8601: Possible null reference assignment.
+                //         o = c.B.A; // 2
+                Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "c.B.A").WithLocation(24, 13),
+                // (26,13): warning CS8601: Possible null reference assignment.
+                //         o = c.B.A; // 3
+                Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "c.B.A").WithLocation(26, 13),
+                // (28,13): warning CS8602: Possible dereference of a null reference.
+                //         o = c.B.A; // 4
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "c.B").WithLocation(28, 13),
+                // (28,13): warning CS8601: Possible null reference assignment.
+                //         o = c.B.A; // 4
+                Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "c.B.A").WithLocation(28, 13),
+                // (30,13): warning CS8602: Possible dereference of a null reference.
+                //         o = c.B.A; // 5
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "c.B").WithLocation(30, 13),
+                // (30,13): warning CS8601: Possible null reference assignment.
+                //         o = c.B.A; // 5
+                Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "c.B.A").WithLocation(30, 13),
+                // (32,13): warning CS8602: Possible dereference of a null reference.
+                //         o = c.B.A; // 6
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "c").WithLocation(32, 13),
+                // (32,13): warning CS8602: Possible dereference of a null reference.
+                //         o = c.B.A; // 6
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "c.B").WithLocation(32, 13),
+                // (32,13): warning CS8601: Possible null reference assignment.
+                //         o = c.B.A; // 6
+                Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "c.B.A").WithLocation(32, 13));
+        }
+
+        [Fact]
+        public void ModifyMembers_03()
+        {
+            var source =
+@"#pragma warning disable 0649
+struct S
+{
+    internal object? F;
+}
+class C
+{
+    internal C? A;
+    internal S B;
+}
+class Program
+{
+    static void M()
+    {
+        object o;
+        C c = new C();
+        o = c.A.A; // 1
+        o = c.B.F; // 1
```

Comment:
>o = c.B.F; // 1 [](start = 8, length = 15)

I would expect a warning on this line as well, but I don't see any.

Reply:
Fixed.